src/wp-includes/rest-api/endpoints/class-wp-rest-font-families-controller.php

```diff
@@ -0,0 +1,554 @@
+	 * @return array                   Decoded array font family settings.
+	 */
+	public function sanitize_font_family_settings( $value ) {
+		// Settings arrive as stringified JSON, since this is a multipart/form-data request.
+		$settings = json_decode( $value, true );
+		$schema   = $this->get_item_schema()['properties']['font_family_settings']['properties'];
+
+		// Sanitize settings based on callbacks in the schema.
+		foreach ( $settings as $key => $value ) {
+			$sanitize_callback = $schema[ $key ]['arg_options']['sanitize_callback'];
+			$settings[ $key ]  = call_user_func( $sanitize_callback, $value );
+		}
+
+		return $settings;
+	}
+
+	/**
+	 * Creates a single font family.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function create_item( $request ) {
+		$settings = $request->get_param( 'font_family_settings' );
+
+		// Check that the font family slug is unique.
+		$query = new WP_Query(
+			array(
+				'post_type'              => $this->post_type,
+				'posts_per_page'         => 1,
+				'name'                   => $settings['slug'],
+				'update_post_meta_cache' => false,
+				'update_post_term_cache' => false,
+			)
+		);
+		if ( ! empty( $query->get_posts() ) ) {
+			return new WP_Error(
+				'rest_duplicate_font_family',
+				/* translators: %s: Font family slug. */
+				sprintf( __( 'A font family with slug "%s" already exists.' ), $settings['slug'] ),
+				array( 'status' => 400 )
+			);
+		}
+
+		return parent::create_item( $request );
+	}
+
+	/**
+	 * Deletes a single font family.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function delete_item( $request ) {
+		$force = isset( $request['force'] ) ? (bool) $request['force'] : false;
+
+		// We don't support trashing for font families.
+		if ( ! $force ) {
+			return new WP_Error(
+				'rest_trash_not_supported',
+				/* translators: %s: force=true */
+				sprintf( __( "Font faces do not support trashing. Set '%s' to delete." ), 'force=true' ),
+				array( 'status' => 501 )
+			);
+		}
+
+		return parent::delete_item( $request );
+	}
+
+	/**
+	 * Prepares a single font family output for response.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param WP_Post         $item    Post object.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response Response object.
+	 */
+	public function prepare_item_for_response( $item, $request ) {
+		$fields = $this->get_fields_for_response( $request );
+		$data   = array();
+
+		if ( rest_is_field_included( 'id', $fields ) ) {
+			$data['id'] = $item->ID;
+		}
+
+		if ( rest_is_field_included( 'theme_json_version', $fields ) ) {
+			$data['theme_json_version'] = 2;
```

Comment:
Is this the right place for a magic number like `2`?

Reply:
I think it makes sense to use `WP_Theme_JSON::LATEST_SCHEMA` here. Added in this commit https://github.com/WordPress/gutenberg/pull/58736/commits/99c98b0a6d653a7f1ccbd42e8021d045afc21ae1 from this PR: https://github.com/WordPress/gutenberg/pull/58736. I'll commit here once that GB PR is merged.

Reply:
We could use `WP_Theme_JSON::LATEST_SCHEMA` here.

But let's say there's a theme.json update and the structure of font families and/or font faces in theme.json are changed. The controllers would need to be updated to support that new structure.

Since we don't know exactly what changes a new theme.json version might bring, I think it's better to manually update the versions that the endpoints accept, after support for a new version is verified.

In the phpunit tests we are [checking to see if the theme_json_version here matches](https://github.com/WordPress/wordpress-develop/pull/6034/files#diff-3075dbacbce6e5a60cdfdd338cea96b6ed5889569c397d7832615c1a43811fafR1001) `WP_Theme_JSON::LATEST_SCHEMA` -- when the latest version of theme.json changes, that would signal an update is needed to these controllers to support the latest version.

Reply:
> But let's say there's a theme.json update and the structure of font families and/or font faces in theme.json are changed. The controllers would need to be updated to support that new structure.

Fair enough! Can we then capture the `2` in a private variable? That way the constant can be documented and link back to the phpunit tests in question.

Reply:
Yes, I think that's a good idea! Updating in Gutenberg to add a constant for each controller, and add an explicit test to check against the latest theme.json version, see https://github.com/WordPress/gutenberg/pull/58736

Reply:
This PR has been updated to use `const LATEST_THEME_JSON_VERSION_SUPPORTED = 2;` in font family/face controllers and in related tests.